### ISSUE TITLE
Proposal: Remove (or clarify) "data" field from ILPError in IL-RFC-3 specification

### ISSUE THREAD

**Issue by sappenin** [interledger/rfcs#188]
The current `ILP Error` format defined in [IL-RFC-3 # ILP Error Format](https://github.com/interledger/rfcs/blob/master/0003-interledger-protocol/0003-interledger-protocol.md#ilp-error-format) contains a `data` attribute which is defined as "up to" 8k of String data, intended for debugging purposes.

I proposed that this be removed from the formal spec for the following reasons:

1. Primarily, this data is unspecified per the specification, which means every implementation will pack this with potentially different debugging data and/or encoding formats, some of which may not even be string-based (despite being encoded to fit into this field as a String).  This will limit the ability of developers who don't control the source of this debug-data to generally understand the data.

1. But let's say the point above is invalid, and that every developer in the world will use human-readable Strings in this field to provide debugging data, I still contend this field should be removed. Consider the following two scenarios:

    In the case where the developer consuming this debug information **does not** control the node that produced the error, this human-readable data is unlikley to be more beneficial than the information provided in the standard error codes.  With or without this field, the recipient of the debug data will need to contact the developer who controls the producer of the debug data, and have them investigate to fix whatever is going wrong.  Additionally, two developers in different organizations should not be reading each other's debug data (see final point).

    In the other case, where the developer controls both sides of the environment (the producer of the debug data _and_ the consumer of the debug data), this developer can and should be inspecting log files or other tools at their disposal to figure out what actually went wrong.
 
1. Last, while security should never rest upon obscurity, leaving this field in the spec will likely lead to implementation errors where developers either misread the spec and include debugging data "because it's in the spec", or worse, developers forget to disable this feature, and inadvertently leak implementation data that might unintentionally provide a vector for remote attacks.

To summarize, this field should be removed because the debug information it provides _should_ be obtained via different means; keeping the field pollutes the standard ILP error specification because many developers won't be able to use the data; and keeping the field potentially weakens the security of the protocol (though, admittedly, this is not really a security weakness assuming implementors don't do anything unwise in their implementations).

**Comment by sappenin** [comment]
HA, so just noticed the fine-print in RFC-3 that says:

> Protocols built on top of ILP that define behavior for certain errors SHOULD specify the encoding format of error data

So, it seems like other protocols (e.g., ILQP?) might specify error data in this field?  If this is the case, I propose we distinguish between a `data` field used for debug information, and a `data` field that is available for standardized sub-protocols.

That said, shouldn't sub-protocols like ILQP be distinct from the ILP Error payload?  Or, put another way, is there a way to make their error messages distinct?  (thinking out loud there).

**Comment by sappenin** [comment]
I'm just catching up to the discussion in [IL RFC PR #180](https://github.com/interledger/rfcs/pull/174#issuecomment-289483204) so I'm going to just close this issue for now and potentially re-open if I still have questions about this field.
